src/js/stores/search.js

```diff
@@ -200,16 +209,15 @@ define(function (require, exports, module) {
             var search = this._registeredSearches[id],
                 types = search.searchHeaders;
             
-            search.searchFilters = [];
-            
             if (types) {
-                search.searchItems = _.reduce(types, function (items, type) {
-                    var options = this._getOptions(type),
-                        filters = this._getFiltersByItems(options, type);
-                    
-                    search.searchFilters.push(filters);
-                    return items.push(options);
-                }.bind(this), Immutable.List());
+                var filterItems = this._getFilterOptions(types),
+                    searchItems = _.reduce(types, function (items, type) {
+                        var options = this._getOptions(type);
+
+                        return items.push(options);
+                    }.bind(this), Immutable.List());
```

Comment:
It would be better to build this with an array and then convert the complete array to an immutable list at the end.